### PR TITLE
Delete robot account on component deletion

### PR DESCRIPTION
**Pull request by mmorhun**
This PR adds functionality to delete robot account which was create on image repository creation.
To make the clean up, a new finalizer `image-repository.component.appstudio.openshift.io/finalizer` is used.

### CODE REVIEW

File: controllers/component_image_controller.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+	"github.com/go-logr/logr"
+	appstudioredhatcomv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/image-controller/pkg/quay"
+)
+
+const (
+	ImageAnnotationName         = "image.redhat.com/image"
+	GenerateImageAnnotationName = "image.redhat.com/generate"
+
+	ImageRepositoryFinalizer = "image-repository.component.appstudio.openshift.io/finalizer"
```
Comment (Michkov) [inline controllers/component_image_controller.go:43]: image-controller.appstudio.openshift.io/finalizer ? It's for component CR so `component` is not necessary in the name, the finalizer should contain name of service which handles it.

File: controllers/component_image_controller.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+	}
+
+	if err := r.Client.Create(ctx, &robotAccountSecret); err != nil {
+		log.Error(err, fmt.Sprintf("error writing robot account token into Secret: %v", robotAccountSecretKey))
+		return ctrl.Result{}, err
+	}
+
+	// Prepare data to update the component with
+	generatedRepository := RepositoryInfo{
+		Image:  fmt.Sprintf("quay.io/%s/%s", r.QuayOrganization, repo.Name),
```
Comment (Michkov) [inline controllers/component_image_controller.go:166]: this is exactly same as `imageURL` so the variable should be used.

**Review by Michkov — COMMENTED** [review]
newly in log: `E0323 11:24:45.065083 1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.24.3/tools/cache/reflector.go:167: Failed to watch *v1.Secret: unknown (get secrets)`

File: controllers/component_image_controller.go (hunk 1 above)
Comment (mmorhun) [reply controllers/component_image_controller.go:43]: ok

**Review by Michkov — APPROVED** [review]

**Pull request merged**